### PR TITLE
Unit tests for uniswap strategy

### PR DESCRIPTION
**Pull request by tomaszslabon**
Depends on https://github.com/keep-network/coverage-pools/pull/148
This PR contains unit tests for contract `SignerBondsUniswapV2` for functions that were not tested in system tests.

### CODE REVIEW

File: test/SignerBondsUniswapV2.test.js [hunk 1]
```diff
@@ -74,6 +74,181 @@ describe("SignerBondsUniswapV2", () => {
...
+      it("should revert", async () => {
+        await expect(
+          signerBondsUniswapV2
+            .connect(thirdParty)
+            .setMaxAllowedPriceImpact(maxAllowedPriceImpact)
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when max allowed price impact too big", () => {
```
Comment (pdyraga) [inline test/SignerBondsUniswapV2.test.js:90]: nitpick: is too high?

File: test/SignerBondsUniswapV2.test.js [hunk 2]
```diff
@@ -74,6 +74,181 @@ describe("SignerBondsUniswapV2", () => {
...
+          signerBondsUniswapV2
+            .connect(governance)
+            .setMaxAllowedPriceImpact(maxAllowedPriceImpact + 1)
+        ).to.be.revertedWith("Maximum value is 10000 basis points")
+      })
+    })
+
+    context("when governance did not set max allowed price impact", () => {
+      const defaultPriceImpact = 100
+      it("should return the default value", async () => {
```
Comment (pdyraga) [inline test/SignerBondsUniswapV2.test.js:102]: This one `describe`s `maxAllowedPriceImpact`, not `setMaxAllowedPriceImpact`. Should we move it to a separate `describe`?

File: test/SignerBondsUniswapV2.test.js [hunk 3]
```diff
@@ -74,6 +74,181 @@ describe("SignerBondsUniswapV2", () => {
...
+      it("should revert", async () => {
+        await expect(
+          signerBondsUniswapV2
+            .connect(thirdParty)
+            .setSlippageTolerance(maxBasisPoints)
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when slippage tolerance too big", () => {
```
Comment (pdyraga) [inline test/SignerBondsUniswapV2.test.js:140]: is too high?

File: test/SignerBondsUniswapV2.test.js [hunk 4]
```diff
@@ -74,6 +74,181 @@ describe("SignerBondsUniswapV2", () => {
...
+            .connect(governance)
+            .setSlippageTolerance(maxBasisPoints + 1)
+        ).to.be.revertedWith("Maximum value is 10000 basis points")
+      })
+    })
+
+    context("when governance did not set slippage tolerance", () => {
+      const defaultSlippageTolerance = 50
+      it("should return the default value", async () => {
+        expect(await signerBondsUniswapV2.slippageTolerance()).to.be.equal(
```
Comment (pdyraga) [inline test/SignerBondsUniswapV2.test.js:153]: Same as for `maxAllowedPriceImpact` - this one describes `slippageTolerance`, not `setSlippageTolerance`.

File: test/SignerBondsUniswapV2.test.js [hunk 5]
```diff
@@ -74,6 +74,181 @@ describe("SignerBondsUniswapV2", () => {
...
+        await expect(
+          signerBondsUniswapV2.connect(thirdParty).setSwapDeadline(10 * 60) // 10 min
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context("when governance did not set default swap deadline", () => {
+      const defaultSwapDeadline = 20 * 60 // 20 min
+      it("should return the default value", async () => {
+        expect(await signerBondsUniswapV2.swapDeadline()).to.be.equal(
```
Comment (pdyraga) [inline test/SignerBondsUniswapV2.test.js:189]: It does describe `swapDeadline`, not `setSwapDeadline`.

File: test/SignerBondsUniswapV2.test.js [hunk 6]
```diff
@@ -74,6 +74,181 @@ describe("SignerBondsUniswapV2", () => {
...
+        ).to.be.revertedWith("Ownable: caller is not the owner")
+      })
+    })
+
+    context(
+      "when governance did not set default value for revert if auction open",
+      () => {
+        const defaultRevertIfAuctionOpen = true
+        it("should return the default value", async () => {
+          expect(await signerBondsUniswapV2.revertIfAuctionOpen()).to.be.equal(
```
Comment (pdyraga) [inline test/SignerBondsUniswapV2.test.js:227]: It does describe `revertIfAuctionOpen`, not `setRevertIfAuctionOpen`.

File: test/SignerBondsUniswapV2.test.js [hunk 7]
```diff
@@ -74,6 +74,181 @@ describe("SignerBondsUniswapV2", () => {
...
+            defaultRevertIfAuctionOpen
+          )
+        })
+      }
+    )
+
+    context("when governance  set revert if auction open", () => {
```
Comment (dimpar) [inline test/SignerBondsUniswapV2.test.js:234]: how about "when governance sets revert flag on opened auction"

File: test/SignerBondsUniswapV2.test.js [hunk 8]
```diff
@@ -74,6 +74,181 @@ describe("SignerBondsUniswapV2", () => {
...
+      const revertIfAuctionOpen = false
+      beforeEach(async () => {
+        expect(
+          await signerBondsUniswapV2
+            .connect(governance)
+            .setRevertIfAuctionOpen(revertIfAuctionOpen)
+        )
+      })
+
+      it("should set revert if auction open", async () => {
```
Comment (dimpar) [inline test/SignerBondsUniswapV2.test.js:244]: "should set revert flag to false"?

File: test/SignerBondsUniswapV2.test.js (hunk 1 above)
Comment (tomaszslabon) [reply test/SignerBondsUniswapV2.test.js]: Fixed in https://github.com/keep-network/coverage-pools/pull/156/commits/7293ab108d709dce06e5db22230eb2d5ad52d647

File: test/SignerBondsUniswapV2.test.js (hunk 3 above)
Comment (tomaszslabon) [reply test/SignerBondsUniswapV2.test.js]: Fixed in https://github.com/keep-network/coverage-pools/pull/156/commits/7293ab108d709dce06e5db22230eb2d5ad52d647.

File: test/SignerBondsUniswapV2.test.js (hunk 7 above)
Comment (tomaszslabon) [reply test/SignerBondsUniswapV2.test.js]: Fixed in https://github.com/keep-network/coverage-pools/pull/156/commits/d62b70a07fbf0da3bf5514c72469571f438f804e.

File: test/SignerBondsUniswapV2.test.js (hunk 2 above)
Comment (tomaszslabon) [reply test/SignerBondsUniswapV2.test.js]: Yes, done in https://github.com/keep-network/coverage-pools/pull/156/commits/5f635a73cfd5eebb12950e447acdec55b28b0cd5.

File: test/SignerBondsUniswapV2.test.js (hunk 4 above)
Comment (tomaszslabon) [reply test/SignerBondsUniswapV2.test.js]: Fixed in https://github.com/keep-network/coverage-pools/pull/156/commits/5f635a73cfd5eebb12950e447acdec55b28b0cd5.

File: test/SignerBondsUniswapV2.test.js (hunk 5 above)
Comment (tomaszslabon) [reply test/SignerBondsUniswapV2.test.js:185]: Fixed in https://github.com/keep-network/coverage-pools/pull/156/commits/5f635a73cfd5eebb12950e447acdec55b28b0cd5.

File: test/SignerBondsUniswapV2.test.js (hunk 6 above)
Comment (tomaszslabon) [reply test/SignerBondsUniswapV2.test.js:225]: Fixed in https://github.com/keep-network/coverage-pools/pull/156/commits/5f635a73cfd5eebb12950e447acdec55b28b0cd5.

**Review by pdyraga — APPROVED** [review]

**Review by pdyraga — APPROVED** [review]

**Pull request merged**